diff --git a/lib/elliptic/ec/index.js b/lib/elliptic/ec/index.js
index a3fe37e3a5f33f1519dbd605c9c8a47a14155deb..e235a1a06925d056e14cdce9b256f5db183b2b7a 100644
--- a/lib/elliptic/ec/index.js
+++ b/lib/elliptic/ec/index.js
@@ -153,7 +153,7 @@ EC.prototype.sign = function sign(msg, key, enc, options) {
   for (var iter = 0; ; iter++) {
     var k = options.k ?
       options.k(iter) :
-      new BN(drbg.generate(this.n.byteLength()));
+      drbg.generate(this.n.byteLength());
     k = this._truncateToN(k, true);
     if (k.cmpn(1) <= 0 || k.cmp(ns1) >= 0)
       continue;
